masa/base/healthcheck.py

```diff
@@ -1,17 +1,38 @@
 import typing
+import requests
 import bittensor as bt
 
 
 class PingMiner(bt.Synapse):
     sent_from: typing.Optional[str]
     is_active: typing.Optional[bool]
+    version: typing.Optional[int]
 
     def deserialize(self) -> str:
         return self.sent_from
 
 
-def forward_ping(synapse: PingMiner) -> PingMiner:
+def forward_ping(synapse: PingMiner, spec_version: int) -> PingMiner:
     synapse.is_active = True
+    synapse.version = spec_version
     bt.logging.info(f"Got ping from {synapse.sent_from}")
+    bt.logging.info(f"Passing back version {synapse.version}")
 
-    return synapse
\ No newline at end of file
+    return synapse
+
+
+def get_external_ip() -> str:
+    """
+    Fetches the external IP address of the machine.
+
+    Returns:
+        str: The external IP address.
+    """
+    try:
+        response = requests.get("https://api.ipify.org?format=json")
+        response.raise_for_status()
+        ip = response.json().get("ip")
+        return ip
+    except requests.RequestException as e:
+        print(f"An error occurred while fetching the external IP: {e}")
+        return None
```

Comment:
Its great to have utils like this, we could move this to a utils folder or similar in the future, look good for now